tests/kafka-connect-jdbc/src/test/java/io/crate/qa/kafka/KafkaJdbcConnectorCrateDBIntegrationTest.java

```diff
@@ -0,0 +1,179 @@
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.*;
+import org.apache.avro.reflect.ReflectData;
+import org.apache.avro.reflect.ReflectDatumWriter;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.hamcrest.core.Is;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.sql.DriverManager;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.function.Supplier;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+public class KafkaJdbcConnectorCrateDBIntegrationTest {
+
+    @Test
+    public void test_kafka_jdbc_connector_cratedb_roundtrip() throws Exception {
+
+        createMetricsTablesInCrateDB();
+        waitUntil(this::setupKafkConnector);
+        waitUntil(this::isKafkaConnectorReady);
+
+        var schema = ReflectData.get().getSchema(Metrics.class);
+
+        var producer = getProducer();
+
+        var m1 = new Metrics(1, 1);
+        var m2 = new Metrics(2, 2);
+        var m3 = new Metrics(3, 3);
+        var m4 = new Metrics(4, 4);
+        var m5 = new Metrics(5, 5);
+
+        producer.send(toProducerRecord(schema, m1));
+        producer.send(toProducerRecord(schema, m2));
+        producer.send(toProducerRecord(schema, m3));
+        producer.send(toProducerRecord(schema, m4));
+        producer.send(toProducerRecord(schema, m5));
+
+        producer.flush();
+        producer.close();
+
+        Thread.sleep(5000);
+
+        var results = fetchResultsFromCrateDB();
+
+        assertThat(results.size(), is(5));
+
+        assertThat(results.get(0), Is.is(m1));
+        assertThat(results.get(1), Is.is(m2));
+        assertThat(results.get(2), Is.is(m3));
+        assertThat(results.get(3), Is.is(m4));
+        assertThat(results.get(4), Is.is(m5));
+
+    }
+
+    ProducerRecord<Object, Object> toProducerRecord(Schema schema, Metrics metric) throws IOException {
+        return new ProducerRecord<>("metrics", toGenericRecord(schema, metric));
+    }
+
+    GenericRecord toGenericRecord(Schema schema, Metrics metric) throws IOException {
+        var datumWriter = new ReflectDatumWriter<>(schema);
+        var outputStream = new ByteArrayOutputStream();
+
+        var encoder = EncoderFactory.get().binaryEncoder(outputStream, null);
+        datumWriter.write(metric, encoder);
+        encoder.flush();
+
+        var datumReader = new GenericDatumReader<GenericRecord>(schema);
+        var decoder = DecoderFactory.get().binaryDecoder(outputStream.toByteArray(), null);
+        return datumReader.read(null, decoder);
```

Comment:
```suggestion
        try (var outputStream = new ByteArrayOutputStream()) {

          var encoder = EncoderFactory.get().binaryEncoder(outputStream, null);
          datumWriter.write(metric, encoder);
          encoder.flush();

          var datumReader = new GenericDatumReader<GenericRecord>(schema);
          var decoder = DecoderFactory.get().binaryDecoder(outputStream.toByteArray(), null);
          return datumReader.read(null, decoder);
        }
```

Reply:
Good point, although there is nothing to close on a `ByteArrayOutputStream`.